frb_example/benches/dart/lib/main.dart

```diff
@@ -0,0 +1,119 @@
+  // on web, env useJSON is unset, but server will send a --json cli arg instead
+  // this is because WASM cannot be fed any env var, but happily takes cli args.
+  final allowJSON = useJSON || args.contains('--json');
+  final log = Logger('frb_example/benches');
+  if (allowJSON) {
+    Logger.root.level = Level.INFO;
+  } else {
+    Logger.root.level = Level.ALL;
+  }
+  Logger.root.onRecord.listen((record) {
+    print('${record.level.name}: ${record.time}: ${record.message}');
+  });
+  log.fine('flutter_rust_bridge_benchmark_suite example program start (dylibPath=$dylibPath)');
+  log.finer('construct api');
+  if (isSilicon) {
+    log.warning("⚠️  [env] SILICON: uses open method for dynamic library");
+  }
+  if (useJSON) {
+    log.info("🗃️  [env] JSON: switch to json output");
+  }
+  if (isWeb && !allowJSON) {
+    log.warning("⚠️  on Web, Rust logs to stdout are redirected to browser's console (or simply choose JSON)");
+  }
+  final api = initializeBenchExternalLibrary(dylibPath, useJSON: allowJSON);
+
+  // here freely do as much setup as you need,
+  // since only calls to bridge-wired functions will record any metric
+  final List<UuidValue> thousandUuids = seed(UuidSeeder(), just_1000);
+  final List<String> thousandStrings = seed(StringSeeder(), just_1000);
+  final List<UuidValue> hundredThousandUuids = seed(UuidSeeder(), just_100_000);
+  final List<String> hundredThousandStrings = seed(StringSeeder(), just_100_000);
+
+  // real game starts here
+  // --- (benchmarks:start) (avoid any initialization or computation)
+  // let's get rid of sync first
+  // lest stdout's output be intermingled
+  // (otherwise feel free to open a PR for opentelemetry)
+  api.handleSyncBool(input: false, hint: 'bool');
+  api.handleSyncI16(input: 42, hint: 'sync i16');
+  api.handleSyncF32(input: 42, hint: 'sync f32');
+  api.handleSyncF64(input: 42, hint: 'sync f64');
+  api.handleSyncI32(input: 42, hint: 'sync i32');
+  api.handleSyncI64(input: 42, hint: 'sync i64');
+  api.handleSyncString(input: "I'm a sync string", hint: 'sync string');
+
+  await api.handleBool(input: false, hint: 'bool');
+  await api.handleI16(input: 42, hint: 'i16');
+  await api.handleF32(input: 42, hint: 'f32');
+  await api.handleF64(input: 42, hint: 'f64');
+  await api.handleI32(input: 42, hint: 'i32');
+  await api.handleI64(input: 42, hint: 'i64');
+  await api.handleString(input: "Not a sync string", hint: 'string');
+
+  await api.handleUuids(ids: thousandUuids, hint: 'reverse 1,000 uuids');
+  await api.handleStrings(strings: thousandStrings, hint: 'reverse 1,000 strings');
+  await api.handleUuidsConvertToStrings(ids: thousandUuids, hint: '1,000 uuids converted to strings');
+  await api.handleUuids(ids: hundredThousandUuids, hint: 'reverse 10,000 uuids');
+  await api.handleStrings(strings: hundredThousandStrings, hint: 'reverse 10,000 strings');
+  await api.handleUuidsConvertToStrings(ids: hundredThousandUuids, hint: '10,000 uuids converted to strings');
```

Comment:
seems that we should repeat, e.g. 100 or 1000 or 10000 times, and get the average, instead of just one call.

Reply:
That's an exercise left to the next contributor ^^

Reply:
Well, the extra loop seems to be key to micro-benchmark. For example,

> Make sure you run it for long enough to be able to measure the results in seconds or (better) tens of seconds

(from https://stackoverflow.com/questions/504103/how-do-i-write-a-correct-micro-benchmark-in-java but that link contains a lot of things unrelated to dart/rust)